pkgs/development/python-modules/POT/default.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, fetchPypi
+, buildPythonPackage
+, numpy
+, scipy
+, cython
+, matplotlib
+, scikit-learn
+, cupy
+, pymanopt
+, autograd
+, pytestCheckHook
+, enableDimensionalityReduction ? false
+, enableGPU ? false
+}:
+
+buildPythonPackage rec {
+  pname = "pot";
+  version = "0.7.0";
+
+  src = fetchPypi {
+    pname = "POT";
+    inherit version;
+    sha256 = "01mdsiv8rlgqzvm3bds9aj49khnn33i523c2cqqrl10zg742pb6l";
+  };
+
+  postPatch = ''
+    substituteInPlace setup.cfg \
+      --replace "--cov-report= --cov=ot" \
+                ""
+  '';
+
+  nativeBuildInputs = [ numpy cython ];
+  propagatedBuildInputs = [ numpy scipy ]
+    ++ lib.optionals enableGPU [ cupy ]
+    ++ lib.optionals enableDimensionalityReduction [ pymanopt autograd ];
+  checkInputs = [ matplotlib scikit-learn pytestCheckHook ];
+
+  preCheck = ''
+    rm -r ot
```

Comment:
Please comment what this is.

Reply:
Python prioritizes importing packages from the current directory even when PYTHONPATH is set. It leads to importing the module from the build directory, not the one installed in nix store. Usually it doesn't matter, but in case of compiled Cython modules they're present only in nix store. One of solutions is to remove the "incorrect" module from the build directory. Relevant discussion: https://discourse.nixos.org/t/cant-import-cythonized-modules-at-checkphase/14207 . Will add a comment to the expression.